src/Jenssegers/Mongodb/Query/Builder.php

```diff
@@ -978,9 +978,13 @@ protected function compileWhereBasic(array $where)
     {
         extract($where);
 
-        // Replace like with a Regex instance.
-        if ($operator == 'like') {
-            $operator = '=';
+        // Replace like or not like with a Regex instance.
+        if (in_array($operator, ['like', 'not like'])) {
+            if (Str::startsWith($operator, 'not')) {
```

Comment:
No need to do string operations here:

```suggestion
            if ($operator === 'not') {
```